### PR TITLE
Start drafting 0.8 changelog

### PR DESCRIPTION
**Pull request by minrk**
Lots of good stuff in this release!

cc @willingc

### CODE REVIEW

File: docs/source/changelog.md [hunk 1]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+      'username': 'name'
+      'state': {}
+  }
+  ```
+
+  This data will be encrypted and requires `JUPYTERHUB_CRYPT_KEY` environment variable to be set
+  and the `Authenticator.enable_auth_state` flag to be True.
+  If these are not set, auth_state returned by the Authenticator will not be stored.
+- There is preliminary support for multiple (named) servers per user in the REST API.
+  There is currently no UI for creating named servers, but they can be created
```
Comment (willingc) [inline docs/source/changelog.md]: I would flip this sentence. Multiple (named) servers can be created via API requests, but there is currently no UI to do so.

File: docs/source/changelog.md [hunk 2]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+  for customizing how user server processes are launched.
+- Add `Authenticator.auto_login` flag for skipping the "Login with..." page explicitly.
+- Add `JupyterHub.hub_connect_ip` configuration
+  for the ip that should be used when connecting to the Hub.
+  This is promoting (and deprecating) `DockerSpawner.hub_ip_connect`
+  for use by all Spawners.
+- Add `Spawner.pre_spawn_hook(spawner)` hook for customizing
+  pre-spawn events.
+- Add `JupyterHub.active_server_limit` and `JupyterHub.concurrent_spawn_limit`
+  for limiting the total number of active users and the number of pending Spawns, respectively.
```
Comment (willingc) [inline docs/source/changelog.md]: Perhaps: ...active users on a hub server...

File: docs/source/changelog.md [hunk 3]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+
 #### Changed
 
+- more arguments to spawners are now passed via environment variables (`.get_env()`)
+  rather than CLI arguments (`.get_args()`)
+- internally generated tokens no longer get extra hash rounds,
+  significantly speeding up authentication.
+  The hash rounds were deemed unnecessary because the tokens were already
+  generated with high entropy.
+- `JUPYTERHUB_API_TOKEN` env is no longer removed during single-user start.
```
Comment (willingc) [inline docs/source/changelog.md]: `JUPYTERHUB_API_TOKEN` env is available at all times (?), and it is no longer removed during...

File: docs/source/changelog.md [hunk 4]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+
 #### Fixed
 
+So many things fixed!
+
+- Collisions are checked when users are renamed
+- Fix bug where OAuth authenticators could not logout users
+  due to being redirected right back through the login process.
+- Errors in config files prevent JupyterHub from starting
+  instead of skipping the configuration.
```
Comment (willingc) [inline docs/source/changelog.md]: configuration and logging a warning (? not sure if this is true. if not something for 0.9)

File: docs/source/changelog.md [hunk 5]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+- Raise 403 error on unauthorized user rather than redirect to login,
+  which could cause redirect loop.
+- Set httponly on cookies because it's prudent.
```
Comment (willingc) [inline docs/source/changelog.md]: `httponly`

File: docs/source/changelog.md [hunk 6]
```diff
@@ -7,12 +7,89 @@ command line for details.
...
+  To do so, the `.authenticate` method should return a dict of the form
+
+  ```python
+  {
+      'username': 'name'
+      'state': {}
+  }
+  ```
+
+  This data will be encrypted and requires `JUPYTERHUB_CRYPT_KEY` environment variable to be set
```
Comment (minrk) [inline docs/source/changelog.md]: We don't! https://github.com/jupyterhub/jupyterhub/issues/1349

File: docs/source/changelog.md (hunk 4 above)
Comment (minrk) [inline docs/source/changelog.md]: Fixed. It's a traitlets flag where errors in config files can either be treated as warnings to be logged or fatal errors. An error in configuration now prevents the Hub from starting, with a (hopefully) informative traceback. In 0.7, an error in configuration would result in the Hub *warning* about failed config and proceeding to launch with default config, which is undesirable. Note should be clearer now.

Comment (willingc) [inline docs/source/changelog.md]: Thanks for the explanation.

**Pull request merged**